### PR TITLE
ZEPPELIN-3364. Followup of ZEPPELIN-3362, improve ZeppelinContext & add more test

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR address the remaining issue of ZEPPELIN-3362, and also add more test for ZeppelinContext.

### What type of PR is it?
[ Improvement ]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3364

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
@felixcheung Mind to help review it ? Thanks

**Comment by zjffdu** [comment]
@prabhjyotsingh Do you mean ZeppelinServer will get the append event after the paragraph is completed ?

File: python/src/main/resources/python/zeppelin_context.py [hunk 1]
```diff
@@ -158,13 +159,12 @@ def show_dataframe(self, df, show_index=False, **kwargs):
...
-        print("%table " + header_buf.read() + body_buf.read()) # +
-        #      ("\n<font color=red>Results are limited by {}.</font>" \
-        #          .format(self.max_result) if limit else "")
-        #)
+        body_buf.seek(0)
+        header_buf.seek(0)
+        print("%table " + header_buf.read() + body_buf.read())
         body_buf.close(); header_buf.close()
+        if exceed_limit:
+            print("%html <font color=red>Results are limited by {}.</font>".format(self.max_result))
```
Comment (felixcheung) [inline python/src/main/resources/python/zeppelin_context.py]: do we need a newline separating %html from before that?

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/plugin/PluginManager.java [hunk 2]
```diff
@@ -60,7 +60,8 @@ public NotebookRepo loadNotebookRepo(String notebookRepoClassName) throws IOExce
             (Class.forName(notebookRepoClassName).newInstance());
         return notebookRepo;
       } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
-        LOGGER.warn("Fail to instantiate notebookrepo from classpath directly:" + notebookRepoClassName, e);
+        LOGGER.warn("Fail to instantiate notebookrepo from classpath directly:"
+            + notebookRepoClassName);
```
Comment (felixcheung) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/plugin/PluginManager.java]: think useful to keep the exception in the log though?

File: python/src/main/resources/python/zeppelin_context.py (hunk 1 above)
Comment (zjffdu) [reply python/src/main/resources/python/zeppelin_context.py]: Not necessary, frontend will separate them correctly.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/plugin/PluginManager.java (hunk 2 above)
Comment (zjffdu) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/plugin/PluginManager.java]: Fixed

**Comment by zjffdu** [comment]
@felixcheung comments are addressed except #2890 (comment) which I plan to do in separate PR as it is not a trivial change I believe.

**Comment by prabhjyotsingh** [comment]
@zjffdu  sure, I've created a new JIRA https://issues.apache.org/jira/browse/ZEPPELIN-3369

**Comment by felixcheung** [comment]
ok thanks

**Comment by zjffdu** [comment]
Will merge if no more comments

**Pull request closed without merging**